pkg/types/gcp/validation/machinepool.go

```diff
@@ -54,20 +54,10 @@ func ValidateMachinePool(platform *gcp.Platform, p *gcp.MachinePool, fldPath *fi
 	return allErrs
 }
 
-// ValidateServiceAccount checks that the service account is only supplied for control plane nodes and during
-// a shared vpn installation.
+// ValidateServiceAccount does not do any checks on the service account since it can be set for all nodes and
+// in non-shared vpn installations.
```

Comment:
So why not remove this check altogether?

Reply:
We can, but I thought it would be more obvious when looking at the change to keep the function and remove the heck, or if we need to add a validation back in the future.